File: ios/Classes/LoggingController.swift
import MapboxMaps
import Foundation

final class LoggingController {
    static private var backend: FLT_LOGGINGLogWriterBackend?
    static private let instance = LoggingController()

    static func setup(_ binaryMessanger: FlutterBinaryMessenger) {
        backend = .init(binaryMessenger: binaryMessanger)
        LogConfiguration.registerLogWriterBackend(forLogWriter: instance)
    }
}

extension LoggingController: LogWriterBackend {
    func writeLog(for level: LoggingLevel, message: String) {
        DispatchQueue.main.async {
            LoggingController.backend?.writeLogLevel(
                level.toFLTLoggingLevel(),
                message: message,
                completion: { _ in }
            )
        }
    }
}

File: mapbox_maps_flutter_mobile/android/src/main/kotlin/com/mapbox/maps/mapbox_maps/LoggingController.kt
package com.mapbox.maps.mapbox_maps

import android.os.Handler
import android.os.Looper
import com.mapbox.common.LogConfiguration
import com.mapbox.common.LoggingLevel
import com.mapbox.maps.mapbox_maps.pigeons.*
import io.flutter.plugin.common.BinaryMessenger

class LoggingController : com.mapbox.common.LogWriterBackend {
  private val handler = Handler(Looper.getMainLooper())

  companion object {
    private var backend: _LogWriterBackendApi? = null
    private val instance = LoggingController()

    fun setup(binaryMessenger: BinaryMessenger) {
      backend = _LogWriterBackendApi(binaryMessenger)
      LogConfiguration.registerLogWriterBackend(instance)
    }
  }

  override fun writeLog(level: LoggingLevel, message: String) {
    handler.post {
      backend?.writeLog(level.toFLTLoggingLevel(), message) { }
    }
  }
}